R/data.table.R

```diff
@@ -1131,11 +1135,13 @@ replace_dot_alias = function(e) {
           if (!all(named_idx <- nzchar(lhs))) {
             # friendly error for common case: trailing terminal comma
             n_lhs = length(lhs)
-            root_name <- if (root == "let") "let" else "`:=`"
+            this_call <- if (root == "let") "let" else "`:=`"
+            for (jj in which(!named_idx)) if (jsub[[jj+1L]] %iscall% ":=")
+              stopf("It looks like you re-used `:=` in argument %d a functional assignment call -- use `=` instead: %s(col1=val1, col2=val2, ...)", jj, this_call)
```

Comment:
this error message looks good to me but I was wondering why we need to repeat it here and above?
is there some way to avoid that repetition?

Reply:
I saw the same thing but nothing simple came to mind for combining across the branches. But I think we _can_ at least use a helper to keep the logic in sync; done now, PTAL.